documentation/creating_a_script_plan.md

```diff
@@ -0,0 +1,369 @@
+# Wrapping a script in a plan
+
+Running commands and scripts with Bolt is awesome, but at some point you might find yourself wanting
+to do more. Wrapping a script in a plan is a great way to:
+- Make the script discoverable in your project. Your teammates or module users can now find the
+  plan that runs a script by running `bolt plan show`, or `Get-BoltPlan` in PowerShell.
+- Parameterize script arguments. Wrapping your script in a plan lets you enforce types for arguments
+  to your script, and documents those parameters. Users can see how to use your script by running
+  `bolt plan show <plan>`, or `Get-BoltPlan -Name <plan>` in PowerShell.
+- Log messages before or after running the script.
+- Include the script as part of more complex orchestration (for example, automatically running
+  some recovery commands on any targets the script fails on).
+
+**NOTE:** The following example uses a Shell script, but the same workflow applies for any scripting
+language.
+
+Follow these steps to turn a script into a plan. This example uses a script that pulls down
+an updated image for every Docker container running in a `docker-compose` project:
+
+_update\_images.sh_
+```
+#!/bin/sh
+for c in `docker-compose ps --services |sort`; do
+  echo "redoing $c"
+  docker-compose rm -s -f $c
+  docker-compose pull $c
+done
+docker-compose up -d --remove-orphans
+```
+
+1. Create a new directory named 'manage_docker' to store your Bolt project. When you initialize 
+   the project, Bolt uses the directory name as the Bolt project name.
+1. Inside the 'manage_docker' directory, create a Bolt project:
+
+   _\*nix shell command_
+   ```shell
+   bolt project init
+   ```
+   _PowerShell cmdlet_
+   ```powershell
+   New-BoltProject
+   ```
+2. Make a `scripts/` directory in your Bolt project.
+3. Place the `update_images.sh` script in the `scripts/` directory.
+4. Create a Bolt plan using `bolt plan new --script`.
+   This command creates a Bolt YAML plan that takes a `targets` parameter, runs the script on the
+   targets, and returns the result from the script run. If you'd prefer to create a Puppet language
+   plan instead of YAML, you can run the same command with the `--pp` option.
+
+   _\*nix shell command_
+   ```
+   bolt plan new manage_docker::update_images --script manage_docker/scripts/update_images.sh
+   ```
+   _PowerShell cmdlet_
+   ```powershell
+   New-BoltPlan -Name manage_docker::update_images -Script manage_docker/scripts/update_images.sh
+   ```
```

Comment:
```suggestion
4. Create a Bolt plan that wraps a script:

   _\*nix shell command_
   '''
   bolt plan new manage_docker::update_images --script manage_docker/scripts/update_images.sh
   '''
   _PowerShell cmdlet_
   '''powershell
   New-BoltPlan -Name manage_docker::update_images -Script manage_docker/scripts/update_images.sh
   '''

   This command creates a Bolt YAML plan that takes a `targets` parameter, runs the script on the
   targets, and returns the result from the script run. If you'd prefer to create a Puppet language
   plan instead of YAML, you can run the same command with the `--pp` or `-Pp` option.
```

Just so we're not preferring the nix command over pwsh. (Also, couldn't add code block ticks to the suggestion. Sorry. 😬 )